integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonMergerRDD.scala

```diff
@@ -589,9 +601,91 @@ class CarbonMergerRDD[K, V](
     // Change string type to support all types
     val sampleRdd = scanRdd
       .map(row => (row.get(0, sparkDataType), null))
+    val sortedRdd = sampleRdd.sortBy(_._1, true)(objectOrdering, classTag[AnyRef])
     val value = new DataSkewRangePartitioner(
-      defaultParallelism, sampleRdd, true)(objectOrdering, classTag[Object])
-    CarbonCompactionUtil.getRangesFromVals(value.rangeBounds, value.minMaxVals)
+      defaultParallelism, sortedRdd, true)(objectOrdering, classTag[Object])
+    if(minVal == null && maxVal == null) {
+      CarbonCompactionUtil
+        .getRangesFromVals(value.rangeBounds, value.rangeBounds(0), value.rangeBounds(1))
+    } else {
+      CarbonCompactionUtil.getRangesFromVals(value.rangeBounds, minVal, maxVal)
+    }
+  }
+
+  def getOverallMinMax(carbonInputSplits: mutable.Seq[CarbonInputSplit],
```

Comment:
What is the use of getting overallminmax of to be compacted segments? what is the issue if we use min/max getting from partitioner?

Reply:
It is not giving exact values for min/max in case of big data. For example, in case for 1-10000, RangePartitioner was storing only a few intermediate values in its ordered set in which the last value is not necessarily the max value. So I had to take the overall min/max for the column.

Reply:
ok, Please move this method to CompactionUtil and add the same comment why we need this method.

Reply:
I removed this method anyway as I found a different way of handling ranges min/max without reading the footer as reading footer would be more costly operation.